src/usePosts.js

```diff
@@ -0,0 +1,82 @@
+import { createContext, useContext, useEffect, useReducer } from 'react';
+import postsReducer from './PostsState';
+import Airtable from 'airtable';
+
+export const PostsContext = createContext();
+
+export const PostsProvider = ({ children }) => {
```

Comment:
The posts provider now has most of the business logic contained in it, like how to handle a search and how to create a new post.